File: projeto_biblioteca/models/nusuario.js
const CRUD = require('./CRUD');
const fs = require('fs');

class Usuario extends CRUD {
    constructor(nome, numero_matricula, endereco, tel, email, historico_emprestimos) {
        super('usuarios.json');
        this.nome = nome;
        this.numero_matricula = numero_matricula;
        this.endereco = endereco;
        this.tel = tel;
        this.email = email;
        this.historico_emprestimos = historico_emprestimos;
    }

    abrir() {
        return this.listar();
    }

    salvar() {
        const dados = this.listar();
        dados.push(this);
        fs.writeFileSync(this.caminhoArquivo, JSON.stringify(dados, null, 2));
    }
}

module.exports = Usuario;

File: projeto_biblioteca/models/nusuario.rb
require 'json'
require_relative 'crud'

class Usuario < Crud
  attr_accessor :nome, :numero_matricula, :endereco, :tel, :email, :senha, :historico_emprestimos

  def initialize(nome, numero_matricula, endereco, tel, email, senha, historico_emprestimos)
    super('dados/usuarios.json')
    @nome = nome
    @numero_matricula = numero_matricula
    @endereco = endereco
    @tel = tel
    @email = email
    @senha = senha
    @historico_emprestimos = historico_emprestimos
  end

  def abrir
    listar
  end

  def salvar
    dados = listar
    dados << { nome: @nome, numero_matricula: @numero_matricula, endereco: @endereco, tel: @tel, email: @email, senha: @senha, historico_emprestimos: @historico_emprestimos }
    super(dados)
  end
end
